test/jdk/java/time/test/java/time/zone/TestZoneRulesProvider.java

```diff
@@ -91,4 +94,48 @@ protected NavigableMap<String, ZoneRules> provideVersions(String zoneId) {
             }
         });
     }
+
+    /**
+     * Tests whether registering a provider twice will still leave it registered.
+     */
+    @Test
+    public void test_registerTwice() {
+        String zone = "MyID";
+        var provider = new ZoneRulesProvider() {
+            @Override
+            protected Set<String> provideZoneIds() {
+                return Set.of(zone);
+            }
+
+            @Override
+            protected ZoneRules provideRules(String zoneId, boolean forCaching) {
+                return null;
+            }
+
+            @Override
+            protected NavigableMap<String, ZoneRules> provideVersions(String zoneId) {
+                return null;
+            }
+        }
```

Comment:
Needs a semi-colon, otherwise would not compile.
Anyway, this inner class can be combined with the one in the test above, and made into a separate class.